src/libNetwork/P2PComm.cpp

```diff
@@ -134,6 +137,15 @@ uint32_t SendJob::writeMsg(const void* buf, int cli_sock, const Peer& from,
     ssize_t n = write(cli_sock, (unsigned char*)buf + written_length,
                       message_length - written_length);
 
+    if (P2PComm::IsHostDownOrUnreachable()) {
```

Comment:
Is errno guaranteed to be for specific write sock attempt? SendJob::writeMsg can be called concurrently by different job handlers.

Reply:
Should be okay I think: https://stackoverflow.com/questions/1694164/is-errno-thread-safe